ratis-netty/src/main/java/org/apache/ratis/netty/server/DataStreamManagement.java

```diff
@@ -326,9 +326,11 @@ static long writeTo(ByteBuf buf, Iterable<WriteOption> options,
     for (ByteBuffer buffer : buf.nioBuffers()) {
       final ReferenceCountedObject<ByteBuffer> wrapped = ReferenceCountedObject.wrap(buffer, buf::retain, buf::release);
       try {
-        byteWritten += channel.write(wrapped);
+        byteWritten += channel.write(wrapped.retain());
```

Comment:
@ivandika3 , we need to pass `ReferenceCountedObject` instead of `ByteBuffer` so that the implementation can also retain the buffer for later use.

Reply:
My mistake, I have updated it accordingly.